src/libsystemd/sd-journal/journal-file.c

```
@@ -1684,6 +1692,16 @@ uint64_t journal_file_hash_table_n_items(Object *o) {
         return (sz - offsetof(Object, hash_table.items)) / sizeof(HashItem);
 }
 
+static void write_entry_array_item(JournalFile *f, Object *o, uint64_t i, uint64_t p) {
+        assert(f);
+        assert(o);
+
+        if (JOURNAL_HEADER_COMPACT(f->header))
+                o->entry_array.compact[i] = htole32(p);
```

please throw an error here if p > UINT32_MAX.

(or, maybe just add an assert, in case this is impossible anyway)